Modules/Core/Common/src/itkMetaDataDictionary.cxx

```diff
@@ -21,24 +21,28 @@ namespace itk
 {
 MetaDataDictionary
 ::MetaDataDictionary()
+  : m_Dictionary( new MetaDataDictionaryMapType() )
 {
-  m_Dictionary = new MetaDataDictionaryMapType;
 }
 
 MetaDataDictionary
 ::~MetaDataDictionary()
 {
-  delete m_Dictionary;
-  m_Dictionary = nullptr;
 }
 
 MetaDataDictionary
 ::MetaDataDictionary(const MetaDataDictionary & old)
+  :  m_Dictionary( new MetaDataDictionaryMapType(*old.m_Dictionary))
 {
-  m_Dictionary = new MetaDataDictionaryMapType;
-  *m_Dictionary = *( old.m_Dictionary );
 }
 
+MetaDataDictionary
+::MetaDataDictionary( MetaDataDictionary && rr)
+  :  m_Dictionary( new MetaDataDictionaryMapType(std::move(*rr.m_Dictionary)))
```

Comment:
I guess you know, dynamic memory allocation (as in `new MetaDataDictionaryMapType`) in a move-operation should _preferably_ be avoided. Wouldn't a defaulted move-constructor (`= default`) just do the right thing?

Reply:
Good catch!